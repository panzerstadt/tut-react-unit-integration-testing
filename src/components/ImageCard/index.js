import React from "react";
import { PropTypes } from "prop-types";

import styles from "./index.module.css";

const ImageCard = ({ src, title, author }) => {
  return (
    <div className={styles.container}>
      <div className={styles.imgContainer}>
        <img src={src} alt="image" />
      </div>
      <div className={styles.textContainer}>
        <h2>{title}</h2>
        <span>{author}</span>
      </div>
    </div>
  );
};

ImageCard.propTypes = {
  src: PropTypes.string.isRequired,
  title: PropTypes.string,
  author: PropTypes.string
};

export default ImageCard;
