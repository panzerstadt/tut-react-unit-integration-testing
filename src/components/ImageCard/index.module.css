.container {
  width: 100%;
  max-width: 400px;
  border-radius: 15px;
  border: 1px solid lightgrey;
  overflow: hidden;
  margin: 50px auto;
  box-shadow: 1px 2px 120px black;

  background-color: white;
}

.imgContainer > img {
  width: 100%;
  min-height: 400px;
  object-fit: cover;
}

.textContainer {
  text-align: left;
  padding: 20px;
  min-height: 100px;
}

.textContainer > h2 {
  font-size: 2.5rem;
  margin: 0 0 10px 0;
}
