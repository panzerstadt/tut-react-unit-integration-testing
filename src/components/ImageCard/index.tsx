import React from "react";

import styles from "./index.module.css";

interface Props {
  src: string;
  title?: string;
  author?: string;
}

const ImageCard: React.FC<Props> = ({ src, title, author }) => {
  return (
    <div className={styles.container}>
      <div className={styles.imgContainer}>
        <img src={src} alt="image" />
      </div>
      <div className={styles.textContainer}>
        <h2>{title}</h2>
        <span>{author}</span>
      </div>
    </div>
  );
};


export default ImageCard;
