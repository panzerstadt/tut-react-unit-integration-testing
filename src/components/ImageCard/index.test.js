import React from "react";
import { shallow } from "enzyme";

import ImageCard from "./index";

describe("ImageCard", () => {
  it("should render correctly", () => {
    const img = require("../../assets/19023.jpeg");
    const component = shallow(<ImageCard src={img} />);

    expect(component).toMatchSnapshot();
  });
});
