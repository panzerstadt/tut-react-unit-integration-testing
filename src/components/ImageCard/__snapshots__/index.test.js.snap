// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageCard should render correctly 1`] = `
<div
  className="container"
>
  <div
    className="imgContainer"
  >
    <img
      alt="image"
      src="19023.jpeg"
    />
  </div>
  <div
    className="textContainer"
  >
    <h2 />
    <span />
  </div>
</div>
`;
